### PR TITLE
chore(tslint): update tslint to 4.x

### PR DESCRIPTION
**Pull request by vicb**
@chuck I did clone tslint, fix the issue and send a PR and then I realize I started from the wrong repo and that it was already fixed in the parent repo.

Anyway this PR should remove the bug.

### CODE REVIEW

**Comment by chuckjaz** [comment]
This doesn't appear to work on CircleCI

**Comment by vicb** [comment]
@chuckjaz PTAL

File: modules/@angular/platform-browser/testing/browser_util.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {NgZone} from '@angular/core';
 import {global} from './facade/lang';
 import {getDOM} from './private_import_platform-browser';
 
-export let browserDetection: BrowserDetection = new BrowserDetection(null);
+let browserDetection: BrowserDetection = new BrowserDetection(null);
```
Comment (chuckjaz) [inline modules/@angular/platform-browser/testing/browser_util.ts]: This breaks the file because classes cannot be used before they are declared (probably worked only as a side-effect of how the `export` was generated). To fix, remove the initializer as it is redundant with the the call to `setup()` on line 75.

**Review by chuckjaz — CHANGES_REQUESTED** [review]

**Comment by vicb** [comment]
@chuckjaz PTYAL

File: modules/@angular/platform-browser/testing/browser_util.ts [hunk 2]
```diff
@@ -135,6 +135,10 @@ export function stringifyElement(el: any /** TODO #9100 */): string {
   return result;
 }
 
+browserDetection = new BrowserDetection(null);
```
Comment (chuckjaz) [inline modules/@angular/platform-browser/testing/browser_util.ts]: Why is this necessary? Line 75 contains `BrowserDetection.setup();` for which `setup()` is `browserDetection = new BrowserDetection(null);` Why do we create two?

**Review by chuckjaz — APPROVED** [review]
